bundles/binding/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/NetatmoBinding.java

```diff
@@ -262,35 +262,34 @@ protected void execute() {
 						case STATIONNAME:
 							for (Device device : oauthCredentials.getStationsDataResponse
 									.getDevices()) {
-								if (stationPosition == null) {
-									DecimalType altitude = DecimalType.ZERO;
-									if (device.getAltitude() != null) {
-										altitude = new DecimalType(Math.round(unitSystem.
-												convertAltitude(device.getAltitude())));
-									}
-									stationPosition = new PointType(
-											new DecimalType(
-													new BigDecimal(device.getLatitude()).setScale(6, BigDecimal.ROUND_HALF_UP)),
-											new DecimalType(new BigDecimal(device.getLongitude()).setScale(6, BigDecimal.ROUND_HALF_UP)),
-											altitude);
-								}
 								if (device.getId().equals(deviceId)) {
+									if (stationPositions.get(device) == null) {
+										DecimalType altitude = DecimalType.ZERO;
+										if (device.getAltitude() != null) {
+											altitude = new DecimalType(Math.round(unitSystem.
+													convertAltitude(device.getAltitude())));
```

Comment:
I would recommend that.  The third DecimalType argument to the PointType constructor must be meters above sea level, not feet or another unit.  Thanks.